app/src/main/java/com/example/sharingang/ui/fragments/DetailedItemFragment.kt

```diff
@@ -198,80 +194,79 @@ class DetailedItemFragment : Fragment() {
     }
 
     private fun initRating(item: Item) {
-        itemViewModel.setRated(item)
-        itemViewModel.rated.observe(viewLifecycleOwner, {
-            val visibility =
-                if (!it && item.sold && currentUserProvider.getCurrentUserId() != null) View.VISIBLE
-                else View.GONE
+        itemViewModel.setReviews(item)
+        itemViewModel.reviews.observe(viewLifecycleOwner, {
+            val visibility = if (it.keys.contains(currentUserProvider.getCurrentUserId())
+                && currentUserProvider.getCurrentUserId() != null
+                && it[currentUserProvider.getCurrentUserId()!!]!!
```

Comment:
Since you call curreentUserProvider.getCurrentUserId() three times, maybe you should make a variable for it.